### PR TITLE
Move test suite from jasmine to mocha+chai

### PR DESCRIPTION
**Pull request by smcjones**
Jasmine required web-test-runner due to karma being deprecated.

web-test-runner's interface with jasmine had several issues, including poor
error reporting. If a test failed, it would silently hang without raising an
exception in the browser.

Mocha + chai allows us to modernize the test suite.

### CODE REVIEW

**Review by pkenjora — APPROVED** [review]
The quotes could be more consistent between single anddouble.

**Review by anaesqueda — APPROVED** [review]
LGTM

File: common/tests/ads_api_types_test.ts [hunk 1]
```diff
@@ -30,70 +31,38 @@ type AssertEqual<TestType, End> = End extends TestType
...
-    transform() {
-      return ['a', { b: 'b' }] as [string, { b: string }];
-    }
-  }
-  it('adjusts underscores to camel cases', () => {
+describe('Ads API Types', function () {
+  it('adjusts underscores to camel cases', function () {
     const test1: AssertEqual<CamelCase<'my_underscore'>, 'myUnderscore'> = true;
-    expect(test1).toBeDefined();
+    expect(test1).not.to.be.undefined;
```
Comment (datumest) [inline common/tests/ads_api_types_test.ts:37]: .to.exist;  // https://www.chaijs.com/api/bdd/#exist, here and below

File: common/tests/ads_api_types_test.ts [hunk 2]
```diff
@@ -30,70 +31,38 @@ type AssertEqual<TestType, End> = End extends TestType
...
   });
 
-  it('does not camel case where not necessary', () => {
+  it('does not camel case where not necessary', function () {
```
Comment (datumest) [inline common/tests/ads_api_types_test.ts:40]: TIL that mocha uses `this` https://mochajs.org/#arrow-functions.

No action required, just documenting.

File: sa360/src/tests/frontend_test.ts
Comment (datumest) [inline sa360/src/tests/frontend_test.ts:1]: Diff is a little hard to read here, but looks OK.

no action req'd.

File: sa360/src/tests/rules_test.ts [hunk 3]
```diff
@@ -176,69 +156,6 @@ export async function generateCampaignPacingTestData(
   }
   return values;
 }
-/**
- *
- * Generates geo test data for the tests below.
- */
-export async function generateUserListTestData(
```
Comment (datumest) [inline sa360/src/tests/rules_test.ts:183]: Was this unused everywhere?

**Review by datumest — CHANGES_REQUESTED** [review]
One nit on the `not.to.be.undefined`, and a question about a deleted method, but otherwise LGTM.

File: sa360/src/tests/rules_test.ts (hunk 3 above)
Comment (smcjones) [reply sa360/src/tests/rules_test.ts:183]: I did a blame and it looks like I wrote the function and it was just never used.

File: sa360/src/tests/frontend_test.ts
Comment (smcjones) [reply sa360/src/tests/frontend_test.ts:1]: I fixed it

File: common/tests/ads_api_types_test.ts (hunk 1 above)
Comment (smcjones) [reply common/tests/ads_api_types_test.ts]: fixed

**Pull request merged**